From 3ad27971102c00ba8ea1c995756263263682fb82 Mon Sep 17 00:00:00 2001
From: Andrew Wang <wang@apache.org>
Date: Wed, 16 Dec 2015 16:21:06 -0800
Subject: [PATCH 1087/2748] HDFS-9300. TestDirectoryScanner.testThrottle() is
 still a little flakey. Contributed by Daniel
 Templeton.

(cherry picked from commit 3c0adac88c3cddb821a732aea02bc068cafd7ea8)
(cherry picked from commit c7cb29defe7a52af0484bfbf526b6be0a4cd1b67)

Change-Id: Ic92f76b7214063c25d7ca97ad02786da5c932b60
---
 .../hdfs/server/datanode/TestDirectoryScanner.java |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/TestDirectoryScanner.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/TestDirectoryScanner.java
index 58d0985..ad4fc04 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/TestDirectoryScanner.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/TestDirectoryScanner.java
@@ -588,7 +588,7 @@ public void testThrottling() throws Exception {
       ratio = 0.0f;
       retries = maxRetries;
 
-      while ((retries > 0) && ((ratio < 3f) || (ratio > 4.5f))) {
+      while ((retries > 0) && ((ratio < 2.75f) || (ratio > 4.5f))) {
         scanner = new DirectoryScanner(dataNode, fds, conf);
         ratio = runThrottleTest(blocks);
         retries -= 1;
@@ -597,7 +597,7 @@ public void testThrottling() throws Exception {
       // Waiting should be about 4x running.
       LOG.info("RATIO: " + ratio);
       assertTrue("Throttle is too restrictive", ratio <= 4.5f);
-      assertTrue("Throttle is too permissive", ratio >= 3.0f);
+      assertTrue("Throttle is too permissive", ratio >= 2.75f);
 
       // Test with more than 1 thread
       conf.setInt(DFSConfigKeys.DFS_DATANODE_DIRECTORYSCAN_THREADS_KEY, 3);
-- 
1.7.9.5

